tests/test_default_qubit.py

```diff
@@ -544,8 +544,24 @@ def test_var_hermitian_rotated_state(self):
                     *np.cos(phi)*(np.sin(theta)-np.cos(theta))+35*np.cos(2*phi)+39)
         self.assertAlmostEqual(var, expected, delta=self.tol)
 
+    def test_var_estimate(self):
+        """Test that variance is estimated and not analytically calculated"""
```

Comment:
All this tests is whether the result is not analytically calculated with 3 shots (doesn't test that variance is being estimated -- in fact, the variance could just be set to -1 for any provided shots value and this test would pass). I'm ok with the test, but perhaps update the docstring/comments to indicate what is strictly being tested here

Reply:
I was inspired to write this tests for two reasons:

a) Before this PR, default qubit would always return analytic variances, even if shots != 0
b) The branch of var that returns the variance for finite shots was not covered before this test. As we don't do stochastic tests I figured that this test would do the job